Tests/MongoSwiftTests/APMTests.swift

```diff
@@ -0,0 +1,68 @@
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+import MongoSwift
+import Nimble
+import NIOConcurrencyHelpers
+import TestsCommon
+
+@available(macOS 10.15, *)
+final class APMTests: MongoSwiftTestCase {
```

Comment:
I mentioned above that I think we'll want to test the "continuation is finished when client is closed" behavior, but I can think of a few other cases we should test as well. 

1) concurrent streams work as expected (start multiple concurrent tasks with streams, do some operations, ensure they both receive events) 
2) [not sure off the top of my head how we might test this / if it's possible but I'll think on it] if a stream gets dropped we don't keep needlessly calling its event handler 
3) the buffering policy is respected (open stream, do > 100 ops, then iterate and ensure we only see 100 events)

Reply:
Agree

1. Wrote a test (with a helper) that kicks off tasks in a for loop, each with its own stream. Each stream's elts are checked against each other. 
2. Not needed
3. Done. The test is a little lax (has less than the number of events passed in, instead of exactly 100) because of some events happening during the loop iteration (esp with load balancers) that led to some weird edge cases of 102 events